Frontend/student-hub/src/components/VerificationModal.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Modal, Button, CircularProgress, TextField } from '@mui/material';
+
+const VerificationModal = ({ isOpen, onClose, onSubmit, onResend, isLoading, verificationCode, handleVerificationCodeChange, modalMode }) => {
+    if (modalMode === 'ResetPassword') {
+        return (
+            <div>
+                <h2>Email Verification Code Has Been Sent To Your Email! Enter the verification Code in the Box and Your New Password!</h2>
+                <form onSubmit={onSubmit}>
+                    <div className="verification-container">
+                        {[...Array(6)].map((_, num) => (
+                            <TextField
+                                key={num}
+                                type="text"
+                                className="verification-input"
+                                maxLength="1"
+                                value={verificationCode[num] || ''}
+                                onChange={(e) => handleVerificationCodeChange(e, num)}
+                                required
+                            />
+                        ))}
+                    </div>
+                </form>
+
+                {isLoading && <CircularProgress color="inherit" />}
+            </div>
+        );
+    }
+
+    return (
+        <Modal
+            open={isOpen}
+            onClose={onClose}
+            style={{ backgroundColor: 'white', color: 'black', border: '1px solid black' }}
+        >
+            <div style={{ backgroundColor: 'white', padding: 20, borderRadius: 10, maxWidth: 400, margin: 'auto', marginTop: '20vh' }}>
```

Comment:
Why is this modal portion within the Verification component? We just want to abstract out the shared part which starts at the verification wrapper (`<div className="verification-container">`)

Right now I wouldn't call this component "reusable", you've just made a class that handles the logic for two use-cases.